### PR TITLE
Upgrade prometheus crds for v0.53 compatibility

### PR DESCRIPTION
**Pull request by freecaykes**
## Description

Update Prometheus CRDs to be compatible with v0.53 of Prometheus Operator and Config Loader 

## For PR author

- [x] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [x] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [x] Milestone set according to targeted release.
- [x] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -2,28 +2,28 @@ version: v1.0
 name: Initial Pipeline
 agent:
   machine:
-    type: e1-standard-8
+    type: e1-standard-4
```
Comment (doublek) [inline .semaphore/semaphore.yml:5]: I think semaphore.yml should be reverted?

Comment (doublek) [inline .semaphore/semaphore.yml:5]: I think changes to semaphore.yml should be reverted?

File: Makefile [hunk 2]
```diff
@@ -304,14 +304,15 @@ cluster-create: $(BINDIR)/kubectl $(BINDIR)/kind
 	while ! KUBECONFIG=$(KUBECONFIG) $(BINDIR)/kubectl get serviceaccount default; do echo "Waiting for default serviceaccount to be created..."; sleep 2; done
 
 ## Deploy CRDs needed for UTs.  CRDs needed by ECK that we don't use are not deployed.
+## k create is used for prometheus as a workaround for https://github.com/prometheus-community/helm-charts/issues/1500
```
Comment (doublek) [inline Makefile:307]: nit `kubectl create`?

File: pkg/components/images.go [hunk 3]
```diff
@@ -22,8 +22,7 @@ const (
 	// point to a testing repo but the init image will be pushed to quay, so having
 	// these separate allows pulling the proper test images for the Tigera components
 	// and Init image when testing.
-	ECKRegistry        = "quay.io/"
-	InitRegistry       = "quay.io/"
-	K8sGcrRegistry     = "gcr.io/"
-	PrometheusRegistry = "quay.io/"
```
Comment (doublek) [inline pkg/components/images.go:28]: @hjiawei I don't think PrometheusRegistry is needed. Can you confirm please.

**Review by doublek — COMMENTED** [review]
Semaphore pipeline changes should be reverted, otherwise changes look good to me. Can you also update the PR description as to why we are introducing the two new `coreos-` "components".

File: config/enterprise_versions.yml [hunk 4]
```diff
@@ -87,6 +87,19 @@ components:
   packetcapture-api:
     image: tigera/packetcapture-api
     version: master
+  # coreos-prometheus holds the version of prometheus built for tigera/prometheus
+  # which prometheus operator uses to validate
```
Comment (doublek) [inline config/enterprise_versions.yml:91]: nit: Missed this one. Looks like a partial comment?

File: pkg/components/enterprise.go [hunk 5]
```diff
@@ -157,8 +162,13 @@ var (
 		Image:   "tigera/prometheus-service",
 	}
 
+	ComponentCoreOSPrometheusAlertmanager = component{
```
Comment (doublek) [inline pkg/components/enterprise.go:165]: nit: wonder if this should be just `ComponentCoreOSAlertmanager`

**Review by doublek — COMMENTED** [review]
Couple of more very minor nits. Sorry I missed them earlier.

**Review by doublek — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**